cfilament/src/API.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#define CFILAMENT_SKIP_DECL
```

Comment:
What's this define for?

Reply:
It disables the definition of the various opaque typedefs in the cfilament.h header so that they can be defined to the actual C++ types for compiling the wrapper implementations themselves. Alternatively, the entirety of API.h could be moved to a __cplusplus guard in the cfilament.h, if desired.